assets/js/modules/analytics-4/components/widgets/TopRecentTrendingPages.js

```diff
@@ -0,0 +1,208 @@
+	};
+
+	return [ reportOptions, dates ];
+};
+function TopRecentTrendingPages( { Widget } ) {
+	const viewOnlyDashboard = useViewOnly();
+
+	const dates = getReportOptions()[ 1 ];
+
+	const reportOptions = getReportOptions()[ 0 ];
+
+	const report = useInViewSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getReport( reportOptions )
+	);
+
+	const error = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getErrorForSelector( 'getReport', [
+			reportOptions,
+		] )
+	);
+
+	const titles = useInViewSelect( ( select ) =>
+		! error && report
+			? select( MODULES_ANALYTICS_4 ).getPageTitles(
+					report,
+					reportOptions
+			  )
+			: undefined
+	);
+
+	const loading = useSelect(
+		( select ) =>
+			! select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
+				'getReport',
+				[ reportOptions ]
+			) || titles === undefined
+	);
+
+	const { rows = [] } = report || {};
+
+	const columns = [
+		{
+			field: 'dimensionValues.0.value',
+			Component: ( { fieldValue } ) => {
+				const url = fieldValue;
+				const title = titles[ url ];
+				// Utilizing `useSelect` inside the component rather than
+				// returning its direct value to the `columns` array.
+				// This pattern ensures that the component re-renders correctly based on changes in state,
+				// preventing potential issues with stale or out-of-sync data.
+				// Note: This pattern is replicated in a few other spots within our codebase.
+				const serviceURL = useSelect( ( select ) => {
+					return ! viewOnlyDashboard
+						? select( MODULES_ANALYTICS_4 ).getServiceReportURL(
+								'all-pages-and-screens',
+								{
+									filters: {
+										unifiedPagePathScreen: url,
+									},
+									dates,
+								}
+						  )
+						: null;
+				} );
+
+				if ( viewOnlyDashboard ) {
+					return <MetricTileTablePlainText content={ title } />;
+				}
+
+				return (
+					<Link
+						href={ serviceURL }
+						title={ title }
+						external
+						hideExternalIndicator
+					>
+						{ title }
+					</Link>
+				);
+			},
+		},
+		{
+			field: 'metricValues.0.value',
+			Component: ( { fieldValue } ) => (
+				<strong>{ numFmt( fieldValue ) }</strong>
+			),
+		},
+	];
+
+	return (
+		<MetricTileTable
+			Widget={ Widget }
+			title={ __( 'Top recent trending pages', 'google-site-kit' ) }
+			loading={ loading }
+			rows={ rows }
+			columns={ columns }
+			ZeroState={ ZeroDataMessage }
+			error={ error }
+			moduleSlug="analytics-4"
+			infoTooltip={ __(
+				'Pages with the most pageviews published in the last 3 days',
+				'google-site-kit'
+			) }
+		/>
+	);
+}
+
+TopRecentTrendingPages.propTypes = {
+	Widget: PropTypes.elementType.isRequired,
+};
+
+export default compose(
+	whenActive( {
+		moduleName: 'analytics-4',
+		FallbackComponent: ConnectGA4CTATileWidget,
+	} ),
+	withCustomDimensions( {
+		dimensions:
+			KEY_METRICS_WIDGETS[ KM_ANALYTICS_TOP_RECENT_TRENDING_PAGES ]
+				.requiredCustomDimensions?.[ 0 ],
+		reportOptions: getReportOptions()[ 0 ],
```

Comment:
Oh, I see, this is why we have the function outside the component. That would be worth noting in a comment at the top of the file, as it's unclear otherwise and unexpected 😄 

Additionally, let's have the function return an object so the `reportOptions` and `dates` are keys rather than using numerical indexes for the returned values. `getReportOptions().reportOptions` is more readable than `getReportOptions()[ 0 ]`.

Even better might be `getReportOptions()` only returning the report options, given the dates are simple and consistent/hardcoded we could either specify them in both the function and the component, or have a separate `getDateRange()` function. 🤔